### PR TITLE
chore: add workflow to deploy docs page

### PR DESCRIPTION
**Pull request by ota-meshi**
This PR adds a workflow for building and deploying docs pages.

### CODE REVIEW

File: .github/workflows/GHPages.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            - name: Checkout
+              uses: actions/checkout@v3
+            - uses: actions/setup-node@v3
+              with:
+                  node-version: 18
+            - name: Install Packages
+              run: npm ci
+            - name: Build docs
+              run: |+
+                  export NODE_OPTIONS=--openssl-legacy-provider
```
Comment (ota-meshi) [inline .github/workflows/GHPages.yml:35]: Currently we need this workaround to use vuepress (v1). If we want to remove the workaround, the simplest replacement would be VuePress (v2) or VitePress, neither of which is GA yet.

https://v2.vuepress.vuejs.org/
https://vitepress.vuejs.org/

**Review by MichaelDeBoey — APPROVED** [review]
What URL is this deploying to?

If possible, I'd like to deploy this on https://eslint-community.github.io/eslint-utils or https://eslint-utils.eslint-community.github.io (it's currently https://eslint-utils.mysticatea.dev), so other repos can have VuePress docs as well

**Comment by ota-meshi** [comment]
This PR change deploys the documentation site to https://eslint-community.github.io/eslint-utils.

> so other repos can have VuePress docs as well

I can also publish docs from other repos as doc sites using VuePress in the same way. However, from now on, I would like to use VitePress over VuePress.

I use VitePress for the following docs sites (recently replaced).
https://ota-meshi.github.io/eslint-plugin-regexp/

File: docs/.vuepress/config.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 "use strict"
 
 module.exports = {
+    base: "/eslint-utils/",
```
Comment (ota-meshi) [inline docs/.vuepress/config.js:4]: I forgot to add this and just added it.

**Comment by MichaelDeBoey** [comment]
@ota-meshi Let's do the change to VitePress in another PR

**Pull request merged**